test/verify.hpp

```diff
@@ -133,6 +133,9 @@ struct square_diff_fn
     template <class T, class U>
     double operator()(T x, U y) const
     {
+        // In case (x, y) = (+/-inf, +/-inf)
+        if(std::isnan(x - y))
+            return 0;
```

Comment:
dangerous condition as sometimes, the output on GPU is nan but right output should be valid number. This condition would make these cases undetected

Reply:
Good review. I have added 2 more contraints to make sure `x` and `y` are not `nan` value.
```suggestion
        // In case (x, y) = (+/-inf, +/-inf)
        if(!std::isnan(x) && !std::isnan(y) && std::isnan(x - y))
            return 0;
```

Reply:
if x and y are both non-nan numbers then its substraction can't be nan (it can be normal, subnormal or infinity). Therefore, I don't think we need to add more conditions at this file.

Reply:
@long10024070 do you think `std::isinf(x) && std::isinf(y)` would be more appropriate for your case

Reply:
> @long10024070 do you think `std::isinf(x) && std::isinf(y)` would be more appropriate for your case

Your proposed constraint return `true` if `x` and `y` are infinity with different sign, which is out of my scope.